### ISSUE TITLE
Support for sharing model weights across multiple TPU cores

### ISSUE THREAD

**Issue by fostiropoulos** [tensorflow/tpu#847]
Currently the model size limit for TPUs is at 16GB which is not sufficient for larger architectures (e.g. transformers) Memory pooling can be very helpful. What are the plans in supporting larger models?

**Comment by fostiropoulos** [comment]
@donglimm If I have an architecture identical to a paper published that I am trying to replicate it would be 24GBs. There are GPUs with this much memory but not the batch scaling of TPUs or FOPS. I also use BF16.  

Currently I have reduced the number of layers to be at 15GB and I am only able to train for a batch size of 8 per TPU core (which because of my weight size it turns out to not be significant memory loss from padding, since they are multiples of 128).

Also if it is not off-topic from this, I have been seeing that a significant amount of memory is allocated to rng-bit-generator more details: [StackOverflow](https://stackoverflow.com/questions/64323526/xla-rng-bit-generator-takes-too-much-memory?noredirect=1#comment113743634_64323526)
